addon/ajax-request.js

```diff
@@ -42,16 +42,17 @@ function isJSONAPIContentType(header) {
   return header.indexOf(JSONAPIContentType) === 0;
 }
 
+let pendingRequestCount = 0;
+
 export default class AjaxRequest {
 
   constructor() {
     this.init();
   }
 
   init() {
-    this.pendingRequestCount = 0;
     if (testing) {
-      Test.registerWaiter(() => this.pendingRequestCount === 0);
+      Test.registerWaiter(() => (pendingRequestCount === 0));
```

Comment:
This [compiles](https://babeljs.io/repl/#?evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-2&experimental=true&loose=false&spec=false&code=Test.registerWaiter(()%20%3D%3E%20(pendingRequestCount%20%3D%3D%3D%200))%3B) to
```js
if (testing) {
  Test.registerWaiter(function () {
    return pendingRequestCount === 0;
  });
}
```
That's totally fine. But when browsers start honoring `=>` natively and we stop compiling, I worry that the callback will hold on to `this` unnecessarily. I don't see anything in the [spec](http://www.ecma-international.org/ecma-262/6.0/#sec-arrow-function-definitions-runtime-semantics-evaluation) that tells me whether this would be true or not. I suspect it _won't_ be a problem since otherwise, every time a programmer did something like `setTimeout(() => { console.log('hi') })` it would leak memory. To be safe, you could write this in ES5 syntax.